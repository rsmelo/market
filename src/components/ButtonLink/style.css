.buttonLink {
  composes: button from "../Button/style.css";
  display: inline-block;
}
