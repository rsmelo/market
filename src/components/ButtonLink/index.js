import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Link } from 'react-router-dom'

import style from './style.css'
import styleButton from '../Button/style.css'

const ButtonLink = ({
  to,
  children,
  color,
  appearance,
}) => {
  const styles = classNames(
    style.buttonLink,
    styleButton[appearance],
    styleButton[`${appearance}-${color}`]
  )

  return (
    <Link
      to={to}
      className={styles}
    >
      {children}
    </Link>
  )
}


ButtonLink.defaultProps = {
  to: null,
  appearance: 'flat',
  color: 'green',
}

ButtonLink.propTypes = {
  to: PropTypes.string,
  appearance: PropTypes.oneOf([
    'flat', 'outline',
  ]),
  color: PropTypes.oneOf([
    'green', 'grey',
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
}

export default ButtonLink
