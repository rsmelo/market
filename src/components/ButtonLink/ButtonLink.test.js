import React from 'react'
import renderer from 'react-test-renderer'

import ButtonLink from './index'

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('ButtonLink', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<ButtonLink>Test</ButtonLink>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
