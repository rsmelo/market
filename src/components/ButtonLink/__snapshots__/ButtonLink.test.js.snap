// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink should render correctly 1`] = `
<Link
  className=""
  to={null}
>
  Test
</Link>
`;
